docs/sketch-project-file.md

```diff
@@ -56,8 +57,14 @@ otherwise below). The available fields are:
   information as `<PLATFORM>`, `<PLATFORM_VERSION>`, and `<3RD_PARTY_PLATFORM_URL>` respectively but for the core
   platform dependency of the main core platform. These fields are optional.
 - `libraries:` is a section where the required libraries to build the project are defined. This section is optional.
-  - `<INDEX_LIB_NAME> (<INDEX_LIB_VERSION>)` represents a library from the Arduino Libraries Index, for example,
-    `MyLib (1.0.0)`.
+  - `<INDEX_LIB_NAME> (<INDEX_LIB_VERSION>)` represents a library from the Arduino Libraries Index that is a direct
+    dependency of the sketch, for example, `MyLib (1.0.0)`.
+  - `dependency: <INDEX_LIB_NAME> (<INDEX_LIB_VERSION>)` represents a library from the Arduino Libraries Index that is
+    an "indirect dependency", i.e., a library that is not used directly by the sketch, but is required by other
+    libraries used by the sketch. For example, if `LibUsedInSketch` requires `UtilLib`, this information can be recorded
+    in the profile with the declaration `dependency: UtilLib (1.3.4)`. During sketch compilation, a dependency library
+    is treated the same as a non-dependency library; the only difference is in automatic library management, where a
+    dependency library can be installed or removed together with the library that uses it.
```

Comment:
```suggestion
    in the profile with the declaration `dependency: UtilLib (1.3.4)`. During sketch compilation, a `dependency` library
    is treated the same as a non-`dependency` library; the only difference is in automatic library management, where a
    `dependency` library can be installed or removed together with the library that uses it.
```

I think that treating "dependency" as prose in this context of sketch compilation makes things too confusing, as in this context all the libraries are dependencies. Adding code markup makes it clear that "dependency" is referring to the specific type of library dependency defined in the build profile.